### PR TITLE
feat(behavior_path_planner): add safety check against dynamic objects for start/goal planner

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

should be merged with the [PR](https://github.com/autowarefoundation/autoware.universe/pull/4873) in autoware_launcher

WIP(explain feature later)

## Related links

WIP

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/5c489496-1d6b-534a-9a70-633e72d1e1c3?project_id=prd_jt)

## Interface changes

Nothing

## Effects on system behavior

When safety checker judge it's not safe, you can't execute start/goal_planner.
safety check for goal_planner
[Screencast from 2023年09月04日 19時16分04秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/32741405/83943439-4cd0-4dc5-a271-9c0a94a41b1b)

safety check for start_planner
[Screencast from 2023年09月04日 20時06分33秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/32741405/8241d629-6df6-44db-8cb9-4b20a82e181d)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp [hunk 1]
```diff
@@ -340,6 +341,21 @@ bool GoalPlannerModule::isExecutionRequested() const
 
 bool GoalPlannerModule::isExecutionReady() const
 {
+  if (status_.pull_over_path->partial_paths.empty()) {
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp:344]: write TODO

File: planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp [hunk 2]
```diff
@@ -340,6 +341,21 @@ bool GoalPlannerModule::isExecutionRequested() const
...
+    return true;
+  }
+
+  if (status_.is_safe_static_objects && parameters_->safety_check_params.enable_safety_check) {
+    utils::start_goal_planner_common::updateEgoPredictedPathParams(
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp:349]: unnecedsary to execute this

File: planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp [hunk 3]
```diff
@@ -923,6 +940,71 @@ TurnSignalInfo StartPlannerModule::calcTurnSignalInfo() const
...
+{
+  start_planner_data_.filtered_objects = filtered_objects;
+  start_planner_data_.target_objects_on_lane = target_objects_on_lane;
+  start_planner_data_.ego_predicted_path = ego_predicted_path;
+}
+
+bool StartPlannerModule::isSafePath() const
+{
+  // TODO(Sugahara): should safety check for backward path later
+  // back_finishedでtrueなら前進、falseなら後退中
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp:955]: delete

**Review by kosuke55 — APPROVED** [review]
LGTM!

**Pull request merged**